src/messaging-component.js

```diff
@@ -0,0 +1,116 @@
+
+const museSdkUrl = 'https://ppsong.c0d3.com/cdn/assets/modal/sdk.js';
+let userId = '';
+let isRendered = false;
+const sevenDays = 6.048e+8;
+
+const debounce = (f, ms) => {
+    let timeoutId;
+    return (...args : $ReadOnlyArray<mixed>) => {
+        clearTimeout(timeoutId);
+        timeoutId = setTimeout(() => f(...args), ms);
+    };
+};
+
+const showExitModal = ({ cartRecovery }) => {
+    // don't show modal if cartRecovery is not enabled
+    if (!cartRecovery) {
+        return false;
+    }
+    // don't show modal if user is identified
+    const email = getCookie('paypal-cr-user');
+    if (email) {
+        return false;
+    }
+    // don't show modal if user has no cart items
+    const cart = JSON.parse(getCookie('paypal-cr-cart') || '{}');
+    if (!cart.items) {
+        return false;
+    }
+    // don't show modal if user has seen in last 7 days
+    const lastSeen = Boolean(getCookie('paypal-cr-lastseen'));
+    if (lastSeen) {
+        return false;
+    }
+    if (!isRendered) {
+        // $FlowFixMe
+        ppq('updateExperience', {
+            command:    'SHOW_OVERLAY',
+            visitorId:  userId
+        });
+        setCookie('paypal-cr-lastseen', 'true', sevenDays);
+        isRendered = true;
+    }
+    return isRendered;
+};
+
+const init = (...args : $ReadOnlyArray<{ cartRecovery : { userId : string } }>) => {
+    const config = args[0];
+    if (config.cartRecovery) {
+        userId = config.cartRecovery.userId;
+    }
+    const exitIntentListener = debounce(e => {
+        // $FlowFixMe
+        if (e.screenY <= 150) {
+            showExitModal(...args);
+        }
+    }, 100);
+    
+    const resetIdle = debounce(() => {
+        showExitModal(...args);
+    }, checkIfMobile() ? 30000 : 300000);
+
+    if (document && document.body) {
+        document.addEventListener('DOMContentLoaded', () => {
+            // $FlowFixMe
+            document.body.addEventListener('mousemove', exitIntentListener);
+            // $FlowFixMe
+            document.body.addEventListener('mousemove', resetIdle);
+            // $FlowFixMe
+            document.body.addEventListener('mousedown', resetIdle);
+            // $FlowFixMe
+            document.body.addEventListener('touchstart', resetIdle);
+            // $FlowFixMe
+            document.body.addEventListener('onclick', resetIdle);
+        });
+    }
+};
+
+export const Messaging = {
+    init
+};
+
+export function setup() {
+    loadJavascript(museSdkUrl);
+    // $FlowFixMe
+    ppq('showExperience', 'https://www.paypalobjects.com/muse/cart-recovery-0.3/', 'body', {
+        sessionId:          'BOND007',
+        variant:            'modal',
+        flow:               'cart-recovery',
+        mobileVariant:      'modal',
+        mobileFlow:         'cart-recovery',
+        isMobileEnabled:    'true',
+        isDesktopEnabled:   'true',
+        mrid:               'FY8HBGU6Y2MWW',
```

Comment:
Do we want to hard-code this and other params here?

Reply:
currently, this is what the modal sdk from muse expects and there's nothing on the roadmap that requires us to change any of the fields, so we do want to hardcode these values in.

Reply:
@songz don't believe we need mrid here...similarly sessionId. Lets connect on these params